### PR TITLE
[onert] Add more unit tests for MaxPool2D op

### PR DESCRIPTION
**Pull request by zetwhite**
This PR adds more test cases(with stride=2 and negative cases) for MaxPool2D.

ONE-DCO-1.0-Signed-off-by: seunghui youn <sseung.youn@samsung.com>
Co-authored-by: ragmani <ragmani0216@gmail.com>

### CODE REVIEW

File: tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/MaxPool2D.test.cc [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+                        ));
+
+    _context->setBackends({"train"});
+    _context->setEpoch(4);
+
+    SUCCEED();
+  }
+}
+
+TEST_F(GenModelTrain, NonTrainableOps_Conv2D_MaxPool2D_Stride2Filter2)
```
Comment (zetwhite) [inline tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/MaxPool2D.test.cc:171]:
```
[==========] Running 4 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 4 tests from GenModelTrain
[ RUN      ] GenModelTrain.NonTrainableOps_Conv2D_MaxPool2D
[       OK ] GenModelTrain.NonTrainableOps_Conv2D_MaxPool2D (14 ms)
[ RUN      ] GenModelTrain.NonTrainableOps_Conv2D_MaxPool2D_Depth1_Filter2
[       OK ] GenModelTrain.NonTrainableOps_Conv2D_MaxPool2D_Depth1_Filter2 (3 ms)
[ RUN      ] GenModelTrain.NonTrainableOps_Conv2D_MaxPool2D_Depth2_Filter2
[       OK ] GenModelTrain.NonTrainableOps_Conv2D_MaxPool2D_Depth2_Filter2 (3 ms)
[ RUN      ] GenModelTrain.NonTrainableOps_Conv2D_MaxPool2D_Stride2Filter2
[       OK ] GenModelTrain.NonTrainableOps_Conv2D_MaxPool2D_Stride2Filter2 (3 ms)
[----------] 4 tests from GenModelTrain (24 ms total)
```

File: tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/MaxPool2D.test.cc [hunk 2]
```diff
@@ -168,5 +168,95 @@ TEST_F(GenModelTrain, NonTrainableOps_Conv2D_MaxPool2D_Depth2_Filter2)
   }
 }
 
-// TODO : Add test case to check stride supporting
-// TODO : Add more negative test cases
+TEST_F(GenModelTrain, NonTrainableOps_Conv2D_MaxPool2D_Stride2Filter2)
+{
```
Comment (zetwhite) [inline tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/MaxPool2D.test.cc:172]:
```
Epoch 1/5
1/1 [==============================] - 0s 239us/step - loss: 25.5000
Epoch 2/5
1/1 [==============================] - 0s 158us/step - loss: 19.2126
Epoch 3/5
1/1 [==============================] - 0s 155us/step - loss: 12.9202
Epoch 4/5
1/1 [==============================] - 0s 134us/step - loss: 9.0784
Epoch 5/5
1/1 [==============================] - 0s 147us/step - loss: 6.6907
```

Comment (zetwhite) [reply tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/MaxPool2D.test.cc:172]: (note) I made a PR, but this code is originally written by @ragmani. 
Most of code is copy-past from https://github.com/ragmani/ONE/blob/3920c734227f8f5aa18cfd994293abb41363f447/tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/MaxPool2D.test.cc

**Review by ragmani — APPROVED** [review]
LGTM

**Review by hseok-oh — APPROVED** [review]
